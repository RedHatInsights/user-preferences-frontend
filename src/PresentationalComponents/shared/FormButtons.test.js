import React from 'react';
import { Form, RendererContext } from '@data-driven-forms/react-form-renderer';
import { render } from '@testing-library/react';
import FormButtonWithSpies from './FormButtons';

describe('FormButtons tests', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Form onSubmit={() => undefined}>
        {() => (
          <RendererContext.Provider
            value={{
              formOptions: {
                renderForm: () => null,
                internalRegisterField: () => undefined,
                internalUnRegisterField: () => undefined,
              },
            }}
          >
            <FormButtonWithSpies
              dirtyFieldsSinceLastSubmit={{ oepnshift: true }}
              submitSucceeded
              pristine
            />
          </RendererContext.Provider>
        )}
      </Form>
    );
    expect(container).toMatchSnapshot();
  });
});
