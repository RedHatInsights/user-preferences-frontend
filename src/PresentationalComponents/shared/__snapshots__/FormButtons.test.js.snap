// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormButtons tests should render correctly 1`] = `
<div>
  <div
    class="pref-shared--buttons pf-m-9-col-on-md"
    id="user-pref__form-buttons"
  >
    <div
      class="pf-c-form__group pf-m-action pref-shared--actions pf-u-px-lg pf-u-py-md"
    >
      <div
        class="pf-c-form__group-control"
      >
        <div
          class="pf-c-form__actions"
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-primary"
            data-ouia-component-id="user-pref__submit-button"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="submit"
          >
            Save
          </button>
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-link"
            data-ouia-component-id="user-pref__reset-button"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
